### PR TITLE
feat: Adding a step two in wizard form

### LINKED ISSUE #4671: Wizard: Simplify UX by separating step 1 into two steps
The wizard step one has increase complexity over time. A number of fields are not necessary to publish an event. We can simplify the user experience by adding a step 2 and moving some of the form elements to step 2. Please implement it as follows:

* Add a step two with the heading "Additional Info and Settings"
* Move the following fields to step 2:
  * Links and Social Media
  * Listing Privacy
  * Add Code of Conduct
  * License of Event Data and Content

Screenshot from 2020-07-30 16-03-00

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes #4671

#### Short description of what this resolves:
Adding a new step in wizard

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- app/components/forms/wizard/other-details-step.js  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057&bid=19933429)

File: app/controllers/events/view/edit/other-details.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  @action
+  save() {
+    this.saveEventDataAndRedirectTo(
+      'events.view.index',
+      ['tickets', 'socialLinks', 'copyright', 'tax', 'stripeAuthorization']
+    );
+  }
+
+  @action
+  async move(direction, data) {
```
Comment (niranjan94) [inline app/controllers/events/view/edit/other-details.js:17]: Codacy found an issue: ['data' is defined but never used.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057)

File: app/components/forms/wizard/other-details-step.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+import { merge, orderBy, filter, find } from 'lodash-es';
+import { licenses } from 'open-event-frontend/utils/dictionary/licenses';
+import { timezones } from 'open-event-frontend/utils/dictionary/date-time';
+import { paymentCountries, paymentCurrencies } from 'open-event-frontend/utils/dictionary/payment';
+import { countries } from 'open-event-frontend/utils/dictionary/demography';
+import FormMixin from 'open-event-frontend/mixins/form';
+import { inject as service } from '@ember/service';
+import EventWizardMixin from 'open-event-frontend/mixins/event-wizard';
+import { protocolLessValidUrlPattern } from 'open-event-frontend/utils/validators';
+import ENV from 'open-event-frontend/config/environment';
```
Comment (niranjan94) [inline app/components/forms/wizard/other-details-step.js:14]: Codacy found an issue: ['ENV' is defined but never used.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057)

File: app/components/forms/wizard/other-details-step.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+          ]
+        }
+      }
+    };
+    // Merging the predetermined rules with the rules for social links.
+    validationRules.fields = merge(validationRules.fields, this.socialLinksValidationRules);
+    return validationRules;
+  },
+
+  actions: {
```
Comment (niranjan94) [inline app/components/forms/wizard/other-details-step.js:127]: Codacy found an issue: [Trailing spaces not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057)

File: app/components/forms/wizard/other-details-step.js [hunk 4]
```diff
@@ -0,0 +1,160 @@
+import Component from '@ember/component';
+import { later } from '@ember/runloop';
```
Comment (niranjan94) [inline app/components/forms/wizard/other-details-step.js:2]: Codacy found an issue: ['later' is defined but never used.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057)

File: app/components/forms/wizard/other-details-step.js [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+import { observer, computed } from '@ember/object';
```
Comment (niranjan94) [inline app/components/forms/wizard/other-details-step.js:3]: Codacy found an issue: ['observer' is defined but never used.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- app/components/forms/wizard/other-details-step.js  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057&bid=19933429)

File: app/controllers/events/view/edit/other-details.js (hunk 1 above)
Comment (niranjan94) [inline app/controllers/events/view/edit/other-details.js:17]: Codacy found an issue: ['data' is defined but never used.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- app/components/forms/wizard/other-details-step.js  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057&bid=19933429)

File: app/controllers/events/view/edit/other-details.js (hunk 1 above)
Comment (niranjan94) [inline app/controllers/events/view/edit/other-details.js:17]: Codacy found an issue: ['data' is defined but never used.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057)

File: app/components/forms/wizard/other-details-step.js [hunk 6]
```diff
@@ -0,0 +1,563 @@
...
+        }
+      }
+    };
+    // Merging the predetermined rules with the rules for social links.
+    validationRules.fields = merge(validationRules.fields, this.socialLinksValidationRules);
+    return validationRules;
+  },
+
+  actions: {
+
```
Comment (niranjan94) [inline app/components/forms/wizard/other-details-step.js:412]: Codacy found an issue: [Trailing spaces not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- app/components/forms/wizard/other-details-step.js  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057&bid=19933429)

File: app/components/forms/wizard/other-details-step.js (hunk 6 above)
Comment (niranjan94) [inline app/components/forms/wizard/other-details-step.js:412]: Codacy found an issue: [Trailing spaces not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057)

**Comment by iamareebjamal** [comment]
Shouldn't there be code deletion from basic details step JS?

**Comment by maze-runnar** [comment]
> Shouldn't there be code deletion from basic details step JS?

yes, I would do that.

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- app/components/forms/wizard/other-details-step.js  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057&bid=19933429)

File: app/components/forms/wizard/other-details-step.js [hunk 7]
```diff
@@ -0,0 +1,563 @@
+import Component from '@ember/component';
+import { later } from '@ember/runloop';
+import { observer, computed } from '@ember/object';
+import moment from 'moment';
```
Comment (niranjan94) [inline app/components/forms/wizard/other-details-step.js:4]: Codacy found an issue: [Trailing spaces not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057)

File: app/components/forms/wizard/basic-details-step.js [hunk 8]
```diff
@@ -404,7 +373,7 @@ export default Component.extend(FormMixin, EventWizardMixin, {
       }
     };
     // Merging the predetermined rules with the rules for social links.
-    validationRules.fields = merge(validationRules.fields, this.socialLinksValidationRules);
+   // validationRules.fields = merge(validationRules.fields, this.socialLinksValidationRules);
```
Comment (niranjan94) [inline app/components/forms/wizard/basic-details-step.js:376]: Codacy found an issue: [Expected indentation of 4 spaces but found 3.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057)

File: app/components/forms/wizard/basic-details-step.js [hunk 9]
```diff
@@ -23,9 +23,9 @@ export default Component.extend(FormMixin, EventWizardMixin, {
 
   deletedTickets: [],
 
-  licenses: computed(function() {
-    return orderBy(licenses, 'name');
-  }),
+  // licenses: computed(function() {
+  //   return orderBy(licenses, 'name');
+  // }),
```
Comment (iamareebjamal) [inline app/components/forms/wizard/basic-details-step.js:28]: Please delete, don't comment

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- app/components/forms/wizard/other-details-step.js  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057&bid=19933429)

File: app/components/forms/wizard/other-details-step.js (hunk 7 above)
Comment (niranjan94) [inline app/components/forms/wizard/other-details-step.js:4]: Codacy found an issue: [Trailing spaces not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057)

File: app/components/forms/wizard/basic-details-step.js [hunk 10]
```diff
@@ -442,7 +407,7 @@ export default Component.extend(FormMixin, EventWizardMixin, {
         event,
         type,
         position,
-        quantity      : 100,
+        quantity      : 100,
```
Comment (niranjan94) [inline app/components/forms/wizard/basic-details-step.js:410]: Codacy found an issue: [Trailing spaces not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057)

File: app/components/forms/wizard/basic-details-step.js [hunk 11]
```diff
@@ -404,7 +369,7 @@ export default Component.extend(FormMixin, EventWizardMixin, {
       }
     };
     // Merging the predetermined rules with the rules for social links.
-    validationRules.fields = merge(validationRules.fields, this.socialLinksValidationRules);
+   // validationRules.fields = merge(validationRules.fields, this.socialLinksValidationRules);
```
Comment (niranjan94) [inline app/components/forms/wizard/basic-details-step.js:372]: Codacy found an issue: [Expected indentation of 4 spaces but found 3.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- app/components/forms/wizard/other-details-step.js  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057&bid=19933429)

File: app/components/forms/wizard/basic-details-step.js (hunk 10 above)
Comment (niranjan94) [inline app/components/forms/wizard/basic-details-step.js:410]: Codacy found an issue: [Trailing spaces not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057)

**Comment by iamareebjamal** [comment]
It seems that validation rules are just copied from basic details to other step. Please remove unneeded code from both steps. I wanted to merge this before including work on online event form because then it'll need to be updated again but going ahead with online form with old UI for now

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/components/forms/wizard/other-details-step.js  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057&bid=19933429)

**Comment by maze-runnar** [comment]
> It seems that validation rules are just copied from basic details to other step. Please remove unneeded code from both steps. I wanted to merge this before including work on online event form because then it'll need to be updated again but going ahead with online form with old UI for now

@iamareebjamal I have cleared the validations and included only required validations now.

File: app/components/forms/wizard/other-details-step.js [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+      const { event } = this.data;
+      const copyright = await this.getOrCreate(event, 'copyright', 'event-copyright');
+      const license = find(licenses, { name });
+      copyright.setProperties({
+        licence    : name,
+        logoUrl    : license.logoUrl,
+        licenceUrl : license.link
+      });
+    },
+     saveDraft() {
```
Comment (niranjan94) [inline app/components/forms/wizard/other-details-step.js:145]: Codacy found an issue: [Expected indentation of 4 spaces but found 5.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- app/components/forms/wizard/other-details-step.js  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057&bid=19933429)

File: app/components/forms/wizard/other-details-step.js [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+          identifier : 'external_event_url',
+          optional   : true,
+          rules      : [
+            {
+              type   : 'regExp',
+              value  : protocolLessValidUrlPattern,
+              prompt : this.l10n.t('Please enter a valid url')
+            }
+          ]
+        },
```
Comment (niranjan94) [inline app/components/forms/wizard/other-details-step.js:126]: Codacy found an issue: [Unexpected trailing comma.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6089057)

**Review by iamareebjamal — APPROVED** [review]

**Comment by mariobehling** [comment]
Works perfectly. Thanks!

I added an issue to add this step to the dashboard as well here https://github.com/fossasia/open-event-frontend/issues/4936

**Comment by mariobehling** [comment]
@maze-runnar Please make a small change. I think it is too small to open an issue for that. So, I am just posting it here.

Change "Extra Details".. to "Extra details".. in the wizard top button, please.

**Comment by snitin315** [comment]
> @maze-runnar Please make a small change. I think it is too small to open an issue for that. So, I am just posting it here.
> 
> Change "Extra Details".. to "Extra details".. in the wizard top button, please.

Just created a follow-up PR. #4941

**Pull request merged**